### ISSUE TITLE
undefined index argv when using in web page

### ISSUE THREAD

**Issue by fiammybe** [eviweb/composer-wrapper#10]
Hi, I tried a simple case where I just take the code from the readme : 

my index.php file contains just this : 

``` include "../vendor/autoload.php";

$wc = \evidev\composer\Wrapper::create();
        $exit_code = $wc->run("update");
        // add more code here for example
        exit($exit_code);
```

I had to play with timeout and memory (I'm running it locally on a XAMPP) and then I get the following error message : 
    Notice: Undefined index: argv in C:\xampp71\htdocs\usecomposer\vendor\eviweb\composer-wrapper\src\evidev\composer\Wrapper.php on line 235

I would like to use composer in the backend to power package management for a CMS using the browser, and I thought that your approach was a nice one.

**Comment by eviweb** [comment]
Hi @fiammybe,
thanks for reporting.
In fact what I understand of your use case is that the wrapper is included in an `index.php` of a web application.
Am I right?
For now, this wrapper should be used in a command line application.
May you please detail a little bit more your use case?
Maybe should I be able to enhance the wrapper to answer your needs.
Thanks

**Comment by fiammybe** [comment]
Hi @eviweb , 
I'm trying to use composer as part of a web page, that is correct. We are rewriting key parts of ImpressCMS , a PHP-based CMS that is aimed primarily at people that do not use command-lines and which runs well on shared hosting. Instead of reinventing the wheel, we would like to use composer as package manager for our ImpressCMS addon packages. Part of that is already working from the command-line, but now I wanted to get that available from a web page in the administration section as well.
Basically, I want to be able to do the following things with Composer : 

- install a package from packagist
- uninstall a package
- update a package 
- list installed packages and their versions
- check for updates from the installed packages

Do you think the composer-wrapper is the best way to do that, taking into account the limited resources (RAM, execution time) our users potentially will have on their shared hosting servers?

**Comment by eviweb** [comment]
Hi @fiammybe,
I'm not sure that the wrapper is the best way to do in your case.
I remember that [bolt/bolt](https://github.com/bolt/bolt) project has included the wrapper for a short time before calling composer directly.
Maybe should you give a look to: 
- https://github.com/bolt/bolt/issues/1767
- https://github.com/bolt/bolt/pull/2138/commits/6a4218b78679357b713e195e025f63db8006212d
- https://github.com/bolt/bolt/pull/2138/commits/b69bdb2945e42be0b2996e5d73a561b5b5ff1e8e

In fact the main interest that provides the wrapper is to handle a pseudo installation of Composer.
The way it works after, is only to delegate requests to Composer. _(see [Wrapper:run()](https://github.com/eviweb/composer-wrapper/blob/master/src/evidev/composer/Wrapper.php#L223))_

I see that ImpressCMS has its own `composer.json` with a `require` section which means that composer is needed to perform the installation of this CMS, isn't it?
So I'm not sure that adding an additional layer like the wrapper is relevant in your case.
Personally I think that like it was done by Bolt, calling Composer directly is a better way to do.

Best regards

**Comment by fiammybe** [comment]
I think you are correct. THanks for the pointers to bolt/bolt, I'm looking into how they did it!
